docs: add Walrus Console beta documentation (#3540)
MystenLabs/walrus: docs/content/console/auth.mdx

---
title: Authentication and Accounts
description: How Walrus Console sign-in works with Google and Apple through Sui zkLogin, how your Pearl wallet is provisioned, and how accounts stay separate per identity.
keywords: [walrus console, sign in, zklogin, pearl wallet, apple, google, accounts]
---

Walrus Console uses the same kind of login you already know using Google or Apple; no wallet, no seed phrase, no crypto setup. Just sign in and start storing data.

:::info

> Walrus Console is available on Mainnet through a closed, invite-only beta. It currently supports Google sign-in, and plans to support Apple soon. For the product model, see the [concepts and overview](./overview).

Reviewer: We support sign in with Apple in Beta

:::

## How sign-in works

Console uses Sui [zkLogin](https://docs.sui.io/concepts/cryptography/zklogin). You authenticate with an identity provider you already have, and Console derives a [Sui address](https://docs.sui.io/concepts/cryptography/transaction-auth/keys-addresses) from that identity without exposing your provider account onchain and without asking you to manage a private key.

## Sign in with Google or Apple

> 1. Visit the [Walrus Console app](https://testnet.harbor.walrus.xyz/).

Reviewer: The new URL for Console is: https://console.walrus.xyz/

2. Choose **Continue with Google** and complete the Google sign-in.
3. Console provisions your account and a [Personal Space](./overview#core-concepts-spaces-buckets-and-files), then takes you to the dashboard.

> Apple sign-in joins soon and works the same way, deriving your Sui address from your Apple identity. It accepts Apple's private email relay, so you can use the **Hide My Email** option.

Reviewer: We’re supporting Sign in with Apple for Beta.

## Your Pearl wallet

> On first sign-in, Console silently provisions a Pearl wallet, the embedded wallet Console manages on your behalf, for your derived Sui address. This wallet holds the storage resources your data uses. Console manages the wallet and paying for transactions for you, so you do not need to fund it or sign transactions to get started. More advanced workflows require explicit signing steps.

Reviewer: > More advanced workflows require explicit signing steps.

We don't have any for beta

---

> title: Sign-in and accounts

Reviewer 2: ```suggestion
title: Authentication and Accounts
```

description: How Walrus Console sign-in works with Google and Apple through Sui zkLogin, how your Pearl wallet is provisioned, and how accounts stay separate per identity.
keywords: [walrus console, sign in, zklogin, pearl wallet, apple, google, accounts]
---

Walrus Console signs you in with a familiar identity provider and provisions everything you
need to start storing data. You do not create a wallet, manage a seed phrase, or hold tokens

> to sign up.

Reviewer 2: ```suggestion
Walrus Console uses the same kind of login you already know using Google or Apple; no wallet, no seed phrase, no crypto setup. Just sign in and start storing data.
```

> :::note

Reviewer 2: ```suggestion
:::info
```

Walrus Console is in an invited beta on Mainnet. Google sign-in is available today, and Apple
sign-in joins at beta. For what happens after you sign in, see the

> [quickstart](./quickstart). For the product model, see the [concepts and overview](./overview).

Reviewer 2: ```suggestion
Walrus Console is available in on Mainnet through a closed, invite-only beta. It currently supports Google sign-in, and plans to support Apple soon. For the product model, see the [concepts and overview](./overview).
```

:::

## How sign-in works

> Console uses Sui zkLogin. You authenticate with an identity provider you already have, and

Reviewer 2: link to sui zklogin docs

> Console derives a Sui address from that identity without exposing your provider account onchain

Reviewer 2: link to sui address docs/definition

and without asking you to manage a private key. The result is a standard Sui address that owns

> your data, backed by a sign-in you already know how to use.

Reviewer 2: ```suggestion
and without asking you to manage a private key. 
```

Reviewer 2: redundant

## Sign in with Google

1. Visit [testnet.harbor.walrus.xyz](https://testnet.harbor.walrus.xyz/).
2. Choose **Continue with Google** and complete the Google sign-in.

> 3. Console provisions your account and a Personal Space, then takes you to the dashboard.

Reviewer 2: need some sort of definition/link to what a "personal space" is

## Sign in with Apple

Apple sign-in joins at beta and works the same way. Console derives your Sui address from your
Apple identity and provisions your account and Personal Space.

Console accepts Apple's private email relay, so you can use Apple's **Hide My Email** option.
Console treats the relay address as your account email and delivers any account email through

> it.

Reviewer 2: if this isn't currently supported, IMO this isn't necessary to add, and also feels redundant since it is the same as Google but just Apple

## Your Pearl wallet

> On first sign-in, Console silently provisions a Pearl wallet for your derived Sui address. This

Reviewer 2: need a link/definition of what a "peal" wallet is

wallet holds the storage resources your data uses. Console sponsors gas and manages the wallet
for you, so you do not fund it or sign transactions to get started. The exceptions are the

> explicit signing steps in the encrypted-bucket flow, where you sign with your own service key.

Reviewer 2: ```suggestion
wallet holds the storage resources your data uses. Console manages the wallet and paying for transactions
for you, so you do not need to fund it or sign transactions to get started. More advanced workflows require explicit signing steps. 
```

:::warning

Each identity provider maps to a separate Console account and a separate Sui address. If you
sign in with Google and later sign in with Apple, you get two independent accounts, not one
merged account. Data stored under one is not visible under the other.

:::

Console shows an account-separation notice the first time you sign in so this is clear before
you store anything. Choose one provider and use it consistently. Linking multiple providers to
a single account is planned for a later release.

## Wallet sign-in

Signing in with an existing Sui wallet is planned for a later release. For now, sign in with
Google or Apple through zkLogin. If you hold assets in a personal Sui wallet, note that data
you store through Console lives under your zkLogin-derived address, which is separate from a
wallet you connect later.

## After you sign in

You land on the dashboard with a Personal Space ready to use. The next step is to mint an API
key and make your first request. Follow the [quickstart](./quickstart) to create an encrypted
bucket and upload a file.

## Next steps

- Continue to the [quickstart](./quickstart) to mint a key and upload your first file.

> - See the [concepts and overview](./overview) for spaces, buckets, and asset types.

Reviewer 2: ```suggestion

```
